docs/dev/news/4928.fix

```diff
@@ -0,0 +1 @@
+Check for correct usage of type characteristics in specifications and other places where they were missing
```

Comment:
If I understand correctly than this PR is a breaking change: it will cause Dafny to error in some locations where it previously did not. Might be good to note that in the release notes.